packages/atomic/src/components/search/result-lists/atomic-result-list/atomic-result-list.tsx

```diff
@@ -158,69 +127,44 @@ export class AtomicResultList
         'Folded results will not render any children for the "atomic-result-list". Please use "atomic-folded-result-list" instead.'
       );
     }
+    this.resultList = buildResultList(this.bindings.engine, {
+      options: {fieldsToInclude: this.mergedFieldsToInclude},
+    });
+    this.resultsPerPage = buildResultsPerPage(this.bindings.engine);
     this.resultListCommon = new ResultListCommon({
+      getNumberOfPlaceholders: () => this.resultsPerPageState.numberOfResults,
```

Comment:
What if instead of having a stateful `ResultListCommon` with a `renderList` function, it was rewritten as a functional component ?

Would that even be possible ? 

If so, we'd be able to get rid of these getter functions in favor of the property values directly.

Another thing is that the result list common requires all those properties because a bunch of them are just passed down (plumbing) to other components, and it's opaque.

Would it be possible to achieve something like this (pseudo code) ?

```typescript
//atomic-result-list.tsx:

render() {
return (
   <ResultListCommon 
      placeholder={<ResultsPlaceholder {...this.getPropsForPlaceholderOnly()} />}  
      results={<Results {...this.getPropsForResultsOnly()} />>} 
      {...this.getPropsForResultListCommonOnly())} >
  </ResultListCommon>
}
```

Reply:
```js
export interface ResultListCommonProps {
  bindings: AnyBindings; // multiple used inside common
  host: HTMLElement; // multiple used inside common
  loadingFlag: string; // used inside common to set and passed down to result to unset
  resultTemplateProvider: ResultTemplateProvider; // extracted all the templating in there, used inside and passed down
  getDisplay(): ResultDisplayLayout; // used for classes & passed down to multiple children
  getDensity(): ResultDisplayDensity; // used for classes & passed down to multiple children
  getImageSize(): ResultDisplayImageSize; // used for classes & passed down to multiple children
  getResultListState(): ResultListCommonState<AnyResult>; // various used inside, passed to multiple children
  getNumberOfPlaceholders(): number; // passed down to children placeholders
  getResultRenderingFunction(): ResultRenderingFunction | null; // passed down to children results (alternative to placeholders
  layoutSelector?: string; // used internally to update breakpoints
  nextNewResultTarget: FocusTargetController; // used internally to handle focusing
}
```